File: src/tests/dividends.py
from src import Statement, PTDividendsFormatter, Trading212Parser, Transaction, AssetBuffer
import pandas as pd
import time

if __name__ == "__main__":
    start_time = time.time()
    parser = Trading212Parser()
    files_path = ["./data/t212_2023.csv", "./data/t212_2024.csv", "./data/t212_2024.csv", "./data/t212_2022.csv"]
    statement = Statement([])
    for file in files_path:
        transactions = parser.parse(file)
        statement.add_transactions(transactions)
        
    statement.fetch_data(asset_buffer=AssetBuffer())
    parsing_time = time.time() - start_time
    
    format_start_time = time.time()
    formatter = PTDividendsFormatter()
    dividends = formatter.format(statement, "2023")
    formatting_time = time.time() - format_start_time
    
    df = pd.DataFrame(dividends)
    print(df)
    
    end_time = time.time()
    elapsed_time = end_time - start_time
    print(f"Parsing Time: {parsing_time} seconds")
    print(f"Formatting Time: {formatting_time} seconds")
    print(f"Total elapsed time: {elapsed_time} seconds")

File: js/tests/dividends.js
import { Statement } from "../models/statement.js";
import { Trading212Parser } from "../parsers/trading212parser.js";
import { AssetBuffer } from "../models/asset.js";
import { PTDividendsFormatter } from "../formatters/pt/irs/dividends_formatter.js";

const startTime = Date.now();
const filesPath = [
  "./data/t212_2023.csv",
  "./data/t212_2024.csv",
  "./data/t212_2024.csv",
  "./data/t212_2022.csv",
];
const statement = new Statement([]);

filesPath.forEach((file) => {
  const parser = new Trading212Parser();
  const transactions = parser.parse(file);
  statement.addTransactions(transactions);
});
var parsingTime;
statement
  .fetchData(new AssetBuffer())
  .then(() => {
    parsingTime = Date.now() - startTime;
  })
  .then(() => {
    const formatter = new PTDividendsFormatter();
    const dividends = formatter.format(statement, "2023");
    return dividends;
  })
  .then((dividends) => {
    const formattingTime = Date.now() - startTime;

    console.log(dividends);
    console.log("Nº de linhas: " + dividends.length);

    const endTime = Date.now();
    const elapsedTime = endTime - startTime;
    console.log(`Parsing Time: ${parsingTime / 1000} seconds`);
    console.log(`Formatting Time: ${formattingTime / 1000} seconds`);
    console.log(`Total elapsed time: ${elapsedTime / 1000} seconds`);
  });